### DISCUSSION TITLE
How do I automate the extraction/exportation of statistics from h5 files to excel files?

### DISCUSSION THREAD

**Discussion by a-m-a-n-d-a** [asreview/asreview#1008] [Q&A] [answered]
Hi,

I am in the midst of trying to figure out how to automate the simulation of my  and exporting the statistics into an excel file. At present. I am able to figure out how to automate the simulation of many files against the different parameters (e.g. query_strategy, balance strategy). However, I am stuck with trying to export the data in the h5 file into an excel file.  

I am trying to figure out the code to automatically export the data statistics (wss 80/90/95/100, rss 5/10 etc) from the h5 file into an excel/csv file. If I understand the process correctly, I have to extract the statistics into a dataframe before exporting the dataframe into an excel file?  I understand that there is the asreview API for data handling and statistic but I am pretty clueless as to how to go about coding this.  Unlike the data visualisation and word clouds section in the documentation (which had examples of code which I could follow), I was unable to find a coding example for exporting the simulation data from h5 files. Could I receive some guidance with regards to this? Any coding examples would be greatly appreciated.

**Comment by Rensvandeschoot** [comment, marked as answer]
Did you try to use the [statistics plug-in](https://github.com/asreview/asreview-statistics)?

**Reply by a-m-a-n-d-a** [reply]
Yes. The issue that I am facing is how to extract the statistics provided by the plugin and export them into an excel file (or any other exportable format) using code.

**Reply by Rensvandeschoot** [reply]
thanks for clarifying! 

In a project we ran simulations for [14 datasets](https://github.com/asreview/paper-guidelines-KIFMS) and we extracted the statistics in a [Jupiter notebook](https://github.com/asreview/paper-guidelines-KIFMS/blob/main/analyses/analyses_guidelines_KIFMS.ipynb), resulting in [excel files](https://github.com/asreview/paper-guidelines-KIFMS/tree/main/analyses/output/tables). 

Is this the example you were looking for

**Reply by a-m-a-n-d-a** [reply]
Yes! This is largely the example I was looking for. The only issue left is the format of the loaded files for analysis. I notice that the loaded files in the example are csv/excel files. Is it possible to load .h5 files instead?

**Reply by Rensvandeschoot** [reply]
In the [readme ](https://github.com/asreview/paper-guidelines-KIFMS#descriptive-dataset-statistics) we refer to a [Python script](https://github.com/asreview/paper-guidelines-KIFMS/blob/main/scripts/merge_descriptives.py):

```
To create descriptive statistics for each dataset run:

sh generate_dataset_characteristics.sh

The results are stored in output/simulation/[NAME_DATASET]/descriptives/*.json, 
are merged into one table (csv and excel) by running python scripts/merge_descriptives.py, 
and stored in output/table/data_descriptives.*.
```

Hope this does the trick :-)

**Reply by a-m-a-n-d-a** [reply]
Yes, this is helpful. Appreciate the help on this matter!
